src/legacy/server/config/transform_deprecations.js

```diff
@@ -106,6 +106,8 @@ const deprecations = [
   rename('xpack.telemetry.config', 'telemetry.config'),
   rename('xpack.telemetry.banner', 'telemetry.banner'),
   rename('xpack.telemetry.url', 'telemetry.url'),
+  rename('ui_metric.enabled', 'usageCollection.uiMetric.enabled'),
+  rename('ui_metric.debug', 'usageCollection.uiMetric.debug'),
```

Comment:
Note that these configs are renamed across two different plugins (`ui_metric` plugin to `usageCollection`). Do I need to do anything here for this one?

Reply:
Depends on which PR lands first. If yours does, you have nothing to do and I will reintegrate these to https://github.com/elastic/kibana/pull/52251. If https://github.com/elastic/kibana/pull/52251 lands first, the file does no longer exist and you will have to add them to `src/core/server/config/deprecation/core_deprecations.ts` instead (using `renameFromRoot` instead of `rename`).

Reply:
@pgayvallet can we merge your PR first (#52251) ? Using this file to rename configs between legacy and NP seems to not work. I get this error running kibana with the old config:

```
 FATAL  Error: Unknown configuration key(s): "usageCollection.uiMetric.enabled". Check for spelling errors and ensure that expected plugins are installed.
```


If i set the new config in the kibana.yml file explicitly it does work however.